lib/Fhp/Model/TanRequestChallengeImage.php

```diff
@@ -25,15 +25,33 @@ public function __construct(Bin $bin)
         $mimeTypeLength = ord($mimeTypeLengthString[0]) * 256 + ord($mimeTypeLengthString[1]);
 
         $this->mimeType = substr($data, 2, $mimeTypeLength);
-        $this->data = substr($data, 2 + $mimeTypeLength + 2);
+
+        $data = substr($data, 2 + $mimeTypeLength);
+
+        $dataLengthString = substr($data, 0, 2);
+        $expectedDataLength = ord($dataLengthString[0]) * 256 + ord($dataLengthString[1]);
+        $isDataLength = strlen($data) - 2;
+
+        if($expectedDataLength != $isDataLength)
+        {
+            // This exception is thrown, if there is an encoding problem
+            // f.e.: the serialized action was saved as a string, but not base64 encoded
+            throw new \RuntimeException(
```

Comment:
I think this should be `\InvalidArgumentException`.

Reply:
The problem is not an invalid argument. It's an encoding problem.

Reply:
Ok, I guess.